### PR TITLE
[frontend][backend]Admin profile

### PR DESCRIPTION
**Pull request by wj461**
http://localhost:5173/admin

### CODE REVIEW

**Comment by huhuhu0420** [comment]
image
who is ben

File: frontend/src/lib/components/PUA/adminDataCard.svelte [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		await fetch(backendPath + '/admin/ban-user/1', {
+			method: 'PUT'
+		});
+	}
+
+	async function UnbanUserById() {
+		await fetch(backendPath + '/admin/unban-user/1', {
+			method: 'PUT'
+		});
+	}
```
Comment (huhuhu0420) [inline frontend/src/lib/components/PUA/adminDataCard.svelte]: admin can only ban admin QQ

File: frontend/src/routes/admin/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	import { backendPath } from '$lib/components/PUA/env';
+
+	import admin_icon from '$lib/assets/admin_icon.svg';
+	import AdminDataCard from '$lib/components/PUA/adminDataCard.svelte';
+
+	async function getOrders() {
+		const resp = await fetch(backendPath + '/admin/get-all-orders');
+		const json = await resp.json();
+		console.log(json);
+		return json;
```
Comment (huhuhu0420) [inline frontend/src/routes/admin/+page.svelte]: too many console log

**Comment by wj461** [comment]
> image who is ben

approve by @NOOBDY

File: frontend/src/lib/components/PUA/adminDataCard.svelte (hunk 1 above)
Comment (wj461) [reply frontend/src/lib/components/PUA/adminDataCard.svelte]: Now userID all is 1
我不是很確定要怎麼在這時候獲取userID
現在這個頁面並沒有userID的資訊

File: frontend/src/routes/admin/+page.svelte (hunk 2 above)
Comment (wj461) [reply frontend/src/routes/admin/+page.svelte]: fix

**Comment by wj461** [comment]
這邊的user權限 我想改成1,2,3去設定權限或是string 這樣不會有身份重複問題

File: frontend/src/lib/components/PUA/adminDataCard.svelte [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	<button class="w-full">
+		<div class="flex items-center justify-between">
+			<div class="flex items-center">
+				{#if type == '0'}
+					<img src={order_icon} alt="" class="my-4 ml-6 flex h-16 w-16" />
+				{:else if type == '1'}
+					<img src={customer_icon} alt="" class="my-4 ml-6 flex h-16 w-16" />
+				{:else}
+					<img src={seller_icon} alt="" class="my-4 ml-6 flex h-16 w-16" />
+				{/if}
```
Comment (NOOBDY) [inline frontend/src/lib/components/PUA/adminDataCard.svelte]: why use magic numbers

File: frontend/src/lib/components/PUA/adminDataCard.svelte [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+<div
+	class:border-PUA-dark-gray={ben}
+	class:border-PUA-red={!ben}
+	class="mx-12 my-3 flex justify-between rounded-2xl border-2 hover:bg-slate-200"
+>
+	<button class="w-full">
+		<div class="flex items-center justify-between">
+			<div class="flex items-center">
+				{#if type == '0'}
```
Comment (NOOBDY) [inline frontend/src/lib/components/PUA/adminDataCard.svelte]: `==` and `===` are different is JS

File: frontend/src/lib/components/PUA/adminDataCard.svelte [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		});
+	}
+
+	function changeButtonStatus() {
+		ben = !ben;
+		if (ben) {
+			UnbanUserById();
+		} else {
+			banUserById();
+		}
```
Comment (NOOBDY) [inline frontend/src/lib/components/PUA/adminDataCard.svelte]: use `async`/`await`

File: frontend/src/routes/admin/+page.svelte [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+			<div class="m-7 flex flex-col justify-between">
+				{#await getUserInfoFunction}
+					<div></div>
+				{:then userInfo}
+					<div>
+						<div class="text-2xl text-PUA-gray">{userInfo.user_name}</div>
+						<div class="text-gray-300">{userInfo.user_email}</div>
+					</div>
+					<div class="text-xl text-PUA-gray">{userInfo.address}</div>
+				{/await}
```
Comment (NOOBDY) [inline frontend/src/routes/admin/+page.svelte]:
```suggestion
				{#await getUserInfoFunction then userInfo}
					<div>
						<div class="text-2xl text-PUA-gray">{userInfo.user_name}</div>
						<div class="text-gray-300">{userInfo.user_email}</div>
					</div>
					<div class="text-xl text-PUA-gray">{userInfo.address}</div>
				{/await}
```

File: frontend/src/routes/admin/+page.svelte [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	let getUserInfoFunction = getUserInfo();
+	let profileTab = 0;
+</script>
+
+<div class="flex justify-center font-bold">
+	<div class="flex w-3/5 flex-col">
+		<div class="flex bg-PUA-stone">
+			<img src={admin_icon} alt="" class="my-6 ml-10 flex h-28 w-28" />
+			<div class="m-7 flex flex-col justify-between">
+				{#await getUserInfoFunction}
```
Comment (NOOBDY) [inline frontend/src/routes/admin/+page.svelte]:
```suggestion
				{#await getUserInfo()}
```

**Review by NOOBDY — CHANGES_REQUESTED** [review]

File: frontend/src/routes/admin/+page.svelte (hunk 7 above)
Comment (wj461) [reply frontend/src/routes/admin/+page.svelte]: fix

File: frontend/src/routes/admin/+page.svelte (hunk 6 above)
Comment (wj461) [reply frontend/src/routes/admin/+page.svelte]: fix

File: frontend/src/lib/components/PUA/adminDataCard.svelte (hunk 5 above)
Comment (wj461) [reply frontend/src/lib/components/PUA/adminDataCard.svelte]: fix

**Comment by wj461** [comment]
> > image who is ben
> 
> approve by @NOOBDY

fix

File: frontend/src/lib/components/PUA/adminDataCard.svelte (hunk 1 above)
Comment (wj461) [reply frontend/src/lib/components/PUA/adminDataCard.svelte]: fix

File: frontend/src/lib/components/PUA/adminDataCard.svelte (hunk 3 above)
Comment (wj461) [reply frontend/src/lib/components/PUA/adminDataCard.svelte]: fix

File: frontend/src/lib/components/PUA/adminDataCard.svelte (hunk 4 above)
Comment (wj461) [reply frontend/src/lib/components/PUA/adminDataCard.svelte]: fix

File: backend/user/repository/postgresql.go [hunk 8]
```diff
@@ -49,7 +49,7 @@ func (p *postgresqlUserRepo) GetByID(ctx context.Context, id int64) (*swagger.Us
 func (p *postgresqlUserRepo) GetAllUser(ctx context.Context) ([]swagger.UserDataShort, error) {
 	sqlStatement := `
 		SELECT 	email, user_id, name, authority, status
-		FROM user_data;
+		FROM user_data WHERE user_id != 1;
 	`
```
Comment (huhuhu0420) [inline backend/user/repository/postgresql.go]: i think it is better that userdata can order by id

File: backend/user/repository/postgresql.go [hunk 9]
```diff
@@ -71,6 +71,60 @@ func (p *postgresqlUserRepo) GetAllUser(ctx context.Context) ([]swagger.UserData
 	return l, nil
 }
 
+func (p *postgresqlUserRepo) GetAllOrder(ctx context.Context) ([]swagger.OrderInfoShort, error) {
+	sqlStatement := `
+		SELECT orders.cart_id, orders.store_id, orders.order_date, orders.user_id, user_data.name  
+		FROM orders LEFT JOIN user_data 
+		ON orders.user_id = user_data.user_id
+	`
```
Comment (huhuhu0420) [inline backend/user/repository/postgresql.go]: it can order by date

**Comment by huhuhu0420** [comment]
image
i think this may be a problem

File: backend/user/repository/postgresql.go (hunk 8 above)
Comment (wj461) [reply backend/user/repository/postgresql.go]: maybe I can 
WHERE authority != 'admin' ?

File: backend/user/repository/postgresql.go (hunk 9 above)
Comment (wj461) [reply backend/user/repository/postgresql.go]: fix

**Review by nonorze-VOXX — COMMENTED** [review]
lgtm

File: backend/user/repository/postgresql.go (hunk 8 above)
Comment (wj461) [reply backend/user/repository/postgresql.go]: fix

**Comment by wj461** [comment]
> image i think this may be a problem

fix

**Review by OuChiaYun — APPROVED** [review]
looks good

**Review by huhuhu0420 — APPROVED** [review]

**Pull request merged**